nixops/resources/vpc_network_interface_attachement.py

```diff
@@ -0,0 +1,202 @@
+        self._client = None
+        self._state = StateDict(depl, id)
+        self.region = self._state.get('region', None)
+        self.handle_create_eni_attachement = Handler(['region', 'networkInterfaceId', 'instanceId', 'deviceIndex' ],
+                                          handle=self.realize_create_eni_attachement)
+
+    def show_type(self):
+        s = super(VPCNetworkInterfaceAttachementState, self).show_type()
+        if self.region: s = "{0} [{1}]".format(s, self.region)
+        return s
+
+    @property
+    def resource_id(self):
+        return self._state.get('attachementId', None)
+
+    def prefix_definition(self, attr):
+        return {('resources', 'vpcNetworkInterfaceAttachements'): attr}
+
+    def get_physical_spec(self):
+        return { 'attachementId': self._state.get('attachementId', None) }
+
+    def get_definition_prefix(self):
+        return "resources.vpcNetworkInterfaceAttachements."
+
+    def connect(self):
+        if self._client: return
+        self._client = nixops.ec2_utils.connect_ec2_boto3(self._state['region'], self.access_key_id)
+
+    def create_after(self, resources, defn):
+        return {r for r in resources if
+                isinstance(r, nixops.resources.vpc_network_interface.VPCNetworkInterfaceState) or
+                isinstance(r, nixops.backends.ec2.EC2State)}
+
+    def create(self, defn, check, allow_reboot, allow_recreate):
+        diff_engine = self.setup_diff_engine(config=defn.config)
+        self.access_key_id = defn.config['accessKeyId'] or nixops.ec2_utils.get_access_key_id()
+        if not self.access_key_id:
+            raise Exception("please set 'accessKeyId', $EC2_ACCESS_KEY or $AWS_ACCESS_KEY_ID")
+
+        for handler in diff_engine.plan():
+            handler.handle(allow_recreate)
+
+        self.ensure_state_up()
+
+    def ensure_state_up(self):
+        config = self.get_defn()
+        self._state["region"] = config["region"]
+        self.connect()
+        if self._state.get('attachementId', None):
+            if self.state != self.UP:
+                self.wait_for_eni_attachement(self._state['networkInterfaceId'])
+
+    def wait_for_eni_attachement(self, eni_id):
+        while True:
+            response = self._client.describe_network_interface_attribute(
+                Attribute='attachment',
+                NetworkInterfaceId=eni_id)
+            if response.get('Attachment', None):
+                if response['Attachment']['Status'] == 'attached':
+                    break
+                elif response['Attachment']['Status'] != "attaching":
+                    raise Exception("eni attachment {0} in an unexpected state {1}".format(
+                        eni_id, response['Attachment']['Status']))
+                self.log_continue(".")
+                time.sleep(1)
+            else:
+                raise Exception("eni {} doesn't have any attachement {}".format(eni_id))
```

Comment:
There are a lot of these `wait_for_X` methods and they all look very similar. Could the commonality be factored out? (We already have `nixops.util.check_wait` for waiting in a loop, see e.g. `wait_for_volume_available`.)

Also `attachement` -> `attachment`.